examples/ws_guessing_game.rs

```diff
@@ -0,0 +1,236 @@
+
+            if len > MAX_LEN {
+                ws.send(FrameType::Text(false), "Request too big".as_bytes())?;
+                ws.send(FrameType::Close, &[])?;
+                return Err(EspError::from_infallible::<ESP_ERR_INVALID_SIZE>());
+            }
+
+            let mut buf = [0; MAX_LEN]; // Small digit buffer can go on the stack
+            ws.recv(buf.as_mut())?;
+            let Ok(user_string) = str::from_utf8(&buf[..len]) else {
+                ws.send(FrameType::Text(false), "[UTF-8 Error]".as_bytes())?;
+                return Ok(());
+            };
+
+            let Some(user_guess) = GuessingGame::parse_guess(user_string) else {
+                ws.send(FrameType::Text(false), "Please enter a number between 1 and 100".as_bytes())?;
+                return Ok(());
+            };
+
+            match session.guess(user_guess) {
+                (Ordering::Greater, n) => {
+                    let reply = format!("Your {} guess was too high", nth(n));
+                    ws.send(FrameType::Text(false), reply.as_ref())?;
+                }
+                (Ordering::Less, n) => {
+                    let reply = format!("Your {} guess was too low", nth(n));
+                    ws.send(FrameType::Text(false), reply.as_ref())?;
+                }
+                (Ordering::Equal, n) => {
+                    let reply = format!(
+                        "You guessed {} on your {} try! Refresh to play again",
+                        session.secret,
+                        nth(n)
+                    );
+                    ws.send(FrameType::Text(false), reply.as_ref())?;
+                    ws.send(FrameType::Close, &[])?;
+                }
+            }
+            Ok::<(), EspError>(())
+        })
+        .unwrap();
+
+    // Keep server running forever
+    core::mem::forget(server);
+
+    // Main task no longer needed, free up some memory
+    Ok(())
+}
+
+fn create_server() -> anyhow::Result<EspHttpServer> {
+    let peripherals = Peripherals::take().unwrap();
+    let sys_loop = EspSystemEventLoop::take()?;
+    let nvs = EspDefaultNvsPartition::take()?;
+
+    let mut wifi = BlockingWifi::wrap(
+        EspWifi::new(peripherals.modem, sys_loop.clone(), Some(nvs))?,
+        sys_loop,
+    )?;
+
+    let wifi_configuration = wifi::Configuration::AccessPoint(AccessPointConfiguration {
+        ssid: SSID.into(),
+        ssid_hidden: true,
+        auth_method: AuthMethod::WPA2Personal,
+        password: PASSWORD.into(),
+        channel: CHANNEL,
+        ..Default::default()
+    });
+    wifi.set_configuration(&wifi_configuration)?;
+    wifi.start()?;
+    wifi.wait_netif_up()?;
+
+    info!(
+        "Created Wi-Fi with WIFI_SSID `{}` and WIFI_PASS `{}`",
+        SSID, PASSWORD
+    );
+
+    let server_configuration = esp_idf_svc::http::server::Configuration {
+        stack_size: STACK_SIZE,
+        ..Default::default()
+    };
+
+    // Keep wifi running forever
+    core::mem::forget(wifi);
```

Comment:
its fine to use mem::forget to make the example more concise and show the need trick with ending the main etc, but in the light that this is an example and many non-trained people will read it - people need to be careful using it. Many people are still not fully aware of the threaded nature, forgetting that they are running on an OS and that its only the underlying C part that is still running. 
What i am try to say is maybe we should add a link to https://doc.rust-lang.org/stable/core/mem/fn.forget.html , and elaborate a small note why its usage is "save" here and people should think before adapting it in there code.

Reply:
Trying to think of a concise way to explain it without too many words. How about

>Keep wifi running forever
>Do not call this if you ever want
>to stop or access it later
>https://doc.rust-lang.org/stable/core/mem/fn.forget.html

?